### ISSUE TITLE
Add config arrays for known and suspicious tokens

### ISSUE THREAD

**Issue by 314159265359879** [blockstack/stacks-wallet-web#5404]
![image](https://github.com/leather-wallet/extension/assets/33360391/fbdaa8bf-96e3-49a1-a6fc-5617d67253d1)

Some tokens currently not in the list that is displayed by default, that perhaps should be.
These other exchanges may feel unfairly treated when we show DIKO and ALEX in the default list but not Velar and STSW.
Velar
SP1Y5YSTAHZ88XYK1VPDH24GY0HPX5J4JECTMY4A1.velar-token
Stackswap
SP1Z92MPDQEWZXW36VX71Q25HKF5K2EPCJ304F275.stsw-token-v4a

Mark said:
> We may need to start an array in the remote configuration file of Stacks addresses that pertain to tokens that we recognize, to augment the list that ALEX recognizes.
> We should add a page to our developer docs that explains our approach and provide [support@leather.io](mailto:support@leather.io) as a contact point for requesting new additions.
> Do you have a sense beyond these two tokens just which others we should include from the start since they're known to us, similar to how we know these ones?

Going through my addresses I found a couple more serious tokens to show in the default list:

Allbridge USDC
SP3Y2ZSH8P7D50B0VBTSX11S7XSG24M1VB9YFQA4K.token-aeusdc
stSTX
SP4SZE494VC2YC5JYG7AYFQ44F5Q4PYV7DVMDPBG.ststx-token
LiSTX 
SM26NBC8SFHNW4P1Y4DFH27974P56WN86C92HPEHH.token-lqstx

For the culture?
NOT
SP32AEEF6WW5Y0NMJ1S8SBSZDAY8R5J32NBZFPKKZ.nope

STX-20 tokens are very cheap to distribute, so it is easy to spam the wallet with those. Right now the default is to display them, I think we should consider moving these to the "show more" list as the default. 

Eventually, the idea is that users can configure this list as they please to show/hide any token in Leather.

**Comment by 314159265359879** [comment]
These are the new contracts for the tokens ALEX migrated:

ALEX: SP102V8P0F7JX67ARQ77WEA3D3CFB5XW39REDT0AM.token-alex::alex

aBTC: SP2XD7417HGPRTREMKF748VNEQPDRR0RMANB7X1NK.token-abtc::bridged-btc

sUSDT: SP2XD7417HGPRTREMKF748VNEQPDRR0RMANB7X1NK.token-susdt::bridged-usdt

sLUNR: SP2XD7417HGPRTREMKF748VNEQPDRR0RMANB7X1NK.token-slunr::bridged-lunr

sSKO: SP2XD7417HGPRTREMKF748VNEQPDRR0RMANB7X1NK.token-ssko::bridged-sko


When users connect on the platform app.alexlab.co or app.xlink.network, they are currently asked to migrate from the legacy tokens to the new ones. The new tokens are currently hidden by default, and users have to click "show more" to reveal them. Lets add them to the list of tokens that are shown by default too.

**Comment by markmhendrickson** [comment]
@kyranjamie would it be simple enough to add an array of addresses to our remote config file for tokens that we recognize in addition to those recognized by the ALEX SDK, which we can check in addition to the SDK when determining whether to show tokens by default?

**Comment by kyranjamie** [comment]
Yeah, this would be https://github.com/leather-wallet/extension/labels/effort%3Asmall

A note on API design for implementer: make an array of objects with a property `address`, rather than just an array of strings. This enables non-breaking API changes when needing to later add additional metadata.

```
[{ address: 'bc1p...' }, { ... }]
```

**Comment by markmhendrickson** [comment]
We would do well to implement two arrays actually: one for "known" tokens and one for "suspicious" tokens.

We could leverage them both in their appropriate ways. E.g. the known tokens to show by default, the suspicious tokens to label as suspicious.

**Comment by pete-watters** [comment]
I'm happy to work on this soon but it seems like a feature for users to manage their own tokens is close to completion in https://github.com/leather-io/extension/pull/5568. 

Does that negate the need to do this?

**Comment by markmhendrickson** [comment]
I'd expect this issue to complement the work on token management in general. In that, this issue's array of known tokens can be leveraged to show a greater range of tokens by default, then the user can decide to manage tokens in a way that hides any of them they choose.

If best to avoid code conflicts, though, we can certainly wait until #5568 gets merged.

**Comment by cbadawi** [comment]
yea agreed.
just seeing this now, a config would make more sense. What i currently did in the manage tokens pr is add a string array `defaultEnabledTokens` in https://github.com/leather-io/extension/blob/15c91b833ac9b8eecfba60800b18d86ff9181c02/src/app/common/utils.ts#L29

**Comment by kyranjamie** [comment]
For known-scam tokens, should we even allow these to be visible in the _See more_ menu, versus hiding entirely?

**Comment by markmhendrickson** [comment]
I'll draft some higher-level thoughts for contextualizing the use of these arrays, but my general sense is we want to take a softer approach to moderation than hiding tokens entirely, even if we view them as "known scam tokens", since there will always remain the possibility of false positives and disputes / differences of opinion as to what qualifies for that designation. So it's better to downgrade and label vs. remove per se.

**Comment by markmhendrickson** [comment]
It appears that many known tokens related to earn opportunities (e.g. liquid Stacking, liquidity pools) are not showing up by default since they aren't available for swapping. They'd benefit from an array of known tokens, perhaps one that is extended by arrays of known tokens provided by trusted app partners.

E.g. see "pool-token" and "APower" below and "stSTXbtc" from https://github.com/leather-io/extension/issues/6171

![Image](https://github.com/user-attachments/assets/cb8a28b4-689c-4493-be62-112815c6895b)

![Image](https://github.com/user-attachments/assets/55b73a01-fe24-4f47-ad56-b8abf4fcff71)
